synapseclient/models/mixins/json_schema.py

```diff
@@ -0,0 +1,467 @@
+                                )
+                                for nce in ce.get("causingExceptions", [])
+                            ],
+                        )
+                        for ce in response.get("validationException", {}).get(
+                            "causingExceptions", []
+                        )
+                    ],
+                ),
+            )
+        return JSONSchemaValidation(
+            object_id=response.get("objectId", ""),
+            object_type=response.get("objectType", ""),
+            object_etag=response.get("objectEtag", ""),
+            id=response.get("schema$id", ""),
+            is_valid=response.get("isValid", ""),
+            validated_on=response.get("validatedOn", ""),
+        )
+
+    async def get_schema_derived_keys_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> JSONSchemaDerivedKeys:
+        """
+        Retrieve derived JSON schema keys for the entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaDerivedKeys: An object containing the derived keys for the entity.
+        """
+        response = await get_json_schema_derived_keys(
+            synapse_id=self.id, synapse_client=synapse_client
+        )
+        return JSONSchemaDerivedKeys(keys=response["keys"])
+
+
+@async_to_sync
+class ContainerEntityJSONSchema(BaseJSONSchema):
+    """
+    Mixin class to provide JSON schema functionality.
+    This class is intended to be used with classes that represent Synapse entities.
+    It provides methods to bind, delete, and validate JSON schemas associated with the entity.
+    """
+
+    async def get_schema_validation_statistics_async(
+        self, *, synapse_client: Optional["Synapse"] = None
+    ) -> JSONSchemaValidationStatistics:
+        """
+        Get validation statistics for a container entity.
+
+        Args:
+            synapse_client (Optional[Synapse], optional): The Synapse client instance. If not provided,
+                the last created instance from the Synapse class constructor will be used.
+
+        Returns:
+            JSONSchemaValidationStatistics: The validation statistics.
+        """
+        response = await get_json_schema_validation_statistics(
+            synapse_id=self.id, synapse_client=synapse_client
+        )
+        return JSONSchemaValidationStatistics(
+            container_id=response.get("containerId", ""),
+            total_number_of_children=response.get("totalNumberOfChildren", ""),
+            number_of_valid_children=response.get("numberOfValidChildren", ""),
+            number_of_invalid_children=response.get("numberOfInvalidChildren", ""),
+            number_of_unknown_children=response.get("numberOfUnknownChildren", ""),
```

Comment:
These are `int` types, so I would expect that a `""` is not a valid default value. I would rather see these be `None` if not in the response.